spec/predicates/release.md

```diff
@@ -0,0 +1,217 @@
+does not provide availability as it does not require the registry to serve an
+attested release artifact that is later determined to be untrustworthy.
+
+These use cases are not hypothetical; both of them are the case today for npm's
+[build provenance feature], which includes a [publish attestation]. Note that
+while npm calls this a publish attestation, calling it a release attestation
+better reflects that it's coming from the package registry. Publishing often
+refers to an author sending content to the registry, as in
+[PyPI's trusted publishers feature].
+
+Perhaps surprisingly, this predicate does not depend on [SLSA Provenance], but
+they are better together. In an ecosystem where some (but not all) packages
+have SLSA Provenance and you query by artifact hash, you can't tell the
+difference between an artifact that has been tampered with and one that does
+not yet have SLSA provenance. If the ecosystem has release attestations, you
+can give authoritative answers to what artifacts make up a given release
+version, and what hashes those artifacts should have.
+
+## Prerequisites
+
+This predicate depends on the [in-toto Attestation Framework], as well as the
+[purl-spec] for identifying packaging ecosystems.
+
+## Model
+
+This predicate is for the final stages of the software supply chain, where
+consumers are looking for attestations that the software has not been tampered
+with during distribution.
+
+If a registry supports immutable releases, there SHOULD be one release
+attestation for a given `predicate.purl`.
+
+Even if the registry does not support immutable releases, the attestation
+subject SHOULD include all the artifacts associated with the release at that
+time; otherwise it will be unclear if an artifact was later removed from a
+release. For example, if a release version has an initial release attestation
+with artifact A, and then later has a release attestation with only artifact B,
+that SHOULD be interpreted as the release version now only containing artifact
+B.
+
+## Schema
+
+```jsonc
+{
+  // Standard attestation fields:
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [{ ... }],
+
+  // Predicate:
+  "predicateType": "https://in-toto.io/attestation/release/v0.1",
+  "predicate": {
+    "purl": <ResourceURI>,
+    "releaseId": "..."
+  }
+}
+```
+
+### Fields
+
+**`subject.name`** string
+
+The filename of the artifact as it would appear on disk.
+
+**`predicate.purl`, required** string ([ResourceURI])
+
+A purl uniquely identifying a specific release name and version from a package
+registry.
+
+**`predicate.releaseId`** string
+
+Stable identifier for a release; this should remain unchanged between release
+versions (e.g. it's associated with urllib3, not urllib3 v2.1.0). This will
+allow users to confirm that a release has moved to a new name, and prevent
+confusion if the old name is re-used. This could be an automatically
+incrementing database key or a randomly generated UUID.
```

Comment:
the UUID would have to be generated by the releaser. In that case, we would need to keep track of how the UUIDs map to each artifact. How would we know if the `releaseID` does not match?